helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientImpl.java

```diff
@@ -121,19 +125,24 @@ protected JsonNode getJsonObject(HttpResponse httpResponse) throws IOException {
   }
 
   @VisibleForTesting
-  protected HttpResponse post(String url, Map<String, String> payloads) throws IOException {
+  protected synchronized HttpResponse post(String url, Map<String, String> payloads) throws IOException {
```

Comment:
You will have problem with locking on this method. It even can blocks per node querying. I think we should lock inside get per partition data.

Reply:
Got it! I've changed the locking at `getParttionsStoppableCheck` method, can you take a look?